### ISSUE TITLE
Envoy 1.7.1 compile

### ISSUE THREAD

**Issue by ekarlso** [octarinesec/ModSecurity-envoy#1]
Hi, I am having some issues compiling this using the provided guide in the README.md with 1.7.1.

```
Starting local Bazel server and connecting to it...
DEBUG: /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/io_bazel_rules_go/proto/def.bzl:138:3: You no longer need to call proto_register_toolchains(), it does nothing
External dependency cache directory /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f
make: Entering directory '/home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps'
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/cares.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/libevent.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/gperftools.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/yaml-cpp.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/benchmark.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/luajit.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/nghttp2.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/zlib.dep.env
make: Leaving directory '/home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps'

real	0m0.045s
user	0m0.015s
sys	0m0.011s
DEBUG: /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy/bazel/repositories.bzl:76:5: Fetching external dependencies...
DEBUG: /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy/bazel/repositories.bzl:82:5: 
DEBUG: /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy/bazel/repositories.bzl:83:5: External dependency cache directory /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f
make: Entering directory '/home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps'
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/cares.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/libevent.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/gperftools.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/yaml-cpp.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/benchmark.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/luajit.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/nghttp2.dep.env
No need to rebuild /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/zlib.dep.env
make: Leaving directory '/home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy_deps'

real	0m0.045s
user	0m0.015s
sys	0m0.011s
DEBUG: /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/envoy/bazel/repositories.bzl:84:5: External dep build exited with return code: 0
WARNING: /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/com_github_grpc_grpc/BUILD:1972:1: in srcs attribute of cc_library rule @com_github_grpc_grpc//:grpc_nanopb: please do not import '@com_github_grpc_grpc//third_party/nanopb:pb_common.c' directly. You should either move the file to this package or depend on an appropriate rule there. Since this rule was created by the macro 'grpc_generate_one_off_targets', the error might have been caused by the macro implementation in /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/com_github_grpc_grpc/bazel/grpc_build_system.bzl:172:12
WARNING: /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/com_github_grpc_grpc/BUILD:1972:1: in srcs attribute of cc_library rule @com_github_grpc_grpc//:grpc_nanopb: please do not import '@com_github_grpc_grpc//third_party/nanopb:pb_decode.c' directly. You should either move the file to this package or depend on an appropriate rule there. Since this rule was created by the macro 'grpc_generate_one_off_targets', the error might have been caused by the macro implementation in /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/com_github_grpc_grpc/bazel/grpc_build_system.bzl:172:12
WARNING: /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/com_github_grpc_grpc/BUILD:1972:1: in srcs attribute of cc_library rule @com_github_grpc_grpc//:grpc_nanopb: please do not import '@com_github_grpc_grpc//third_party/nanopb:pb_encode.c' directly. You should either move the file to this package or depend on an appropriate rule there. Since this rule was created by the macro 'grpc_generate_one_off_targets', the error might have been caused by the macro implementation in /home/master/.cache/bazel/_bazel_master/dade022b46ca8a8ec61a5f966e5c06df/external/com_github_grpc_grpc/bazel/grpc_build_system.bzl:172:12
INFO: Analysed target //http-filter-modsecurity:envoy (290 packages loaded).
INFO: Found 1 target...
ERROR: /home/master/ModSecurity-envoy/http-filter-modsecurity/BUILD:13:1: Linking of rule '//http-filter-modsecurity:envoy' failed (Exit 1)
/usr/bin/ld.gold: error: http-filter-modsecurity/libmodsecurity.a:1:9: syntax error, unexpected STRING
/usr/bin/ld.gold: error: http-filter-modsecurity/libmodsecurity.a: not an object or archive
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilterConfig::HttpModSecurityFilterConfig(modsecurity::Decoder const&): error: undefined reference to 'modsecurity::ModSecurity::ModSecurity()'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilterConfig::HttpModSecurityFilterConfig(modsecurity::Decoder const&): error: undefined reference to 'modsecurity::ModSecurity::setConnectorInformation(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilterConfig::HttpModSecurityFilterConfig(modsecurity::Decoder const&): error: undefined reference to 'modsecurity::ModSecurity::setServerLogCb(void (*)(void*, void const*), int)'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilterConfig::HttpModSecurityFilterConfig(modsecurity::Decoder const&): error: undefined reference to 'modsecurity::Rules::loadFromUri(char const*)'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilter::HttpModSecurityFilter(std::shared_ptr<Envoy::Http::HttpModSecurityFilterConfig>): error: undefined reference to 'modsecurity::Transaction::Transaction(modsecurity::ModSecurity*, modsecurity::Rules*, void*)'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilter::~HttpModSecurityFilter(): error: undefined reference to 'modsecurity::Transaction::~Transaction()'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilterConfig::~HttpModSecurityFilterConfig(): error: undefined reference to 'modsecurity::ModSecurity::~ModSecurity()'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilter::onDestroy(): error: undefined reference to 'modsecurity::Transaction::processLogging()'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilter::decodeHeaders(Envoy::Http::HeaderMap&, bool)::{lambda(Envoy::Http::HeaderEntry const&, void*)#1}::operator()(Envoy::Http::HeaderEntry const&, void*) const: error: undefined reference to 'modsecurity::Transaction::addRequestHeader(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilter::decodeHeaders(Envoy::Http::HeaderMap&, bool): error: undefined reference to 'modsecurity::Transaction::processURI(char const*, char const*, char const*)'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilter::decodeHeaders(Envoy::Http::HeaderMap&, bool): error: undefined reference to 'modsecurity::Transaction::processRequestHeaders()'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilter::decodeData(Envoy::Buffer::Instance&, bool): error: undefined reference to 'modsecurity::Transaction::appendRequestBody(unsigned char const*, unsigned long)'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilter::decodeData(Envoy::Buffer::Instance&, bool): error: undefined reference to 'modsecurity::Transaction::processRequestBody()'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilter::encodeHeaders(Envoy::Http::HeaderMap&, bool)::{lambda(Envoy::Http::HeaderEntry const&, void*)#1}::operator()(Envoy::Http::HeaderEntry const&, void*) const: error: undefined reference to 'modsecurity::Transaction::addResponseHeader(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilter::encodeHeaders(Envoy::Http::HeaderMap&, bool): error: undefined reference to 'modsecurity::Transaction::processResponseHeaders(int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilter::encodeData(Envoy::Buffer::Instance&, bool): error: undefined reference to 'modsecurity::Transaction::appendResponseBody(unsigned char const*, unsigned long)'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function Envoy::Http::HttpModSecurityFilter::encodeData(Envoy::Buffer::Instance&, bool): error: undefined reference to 'modsecurity::Transaction::processResponseBody()'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function modsecurity::debug_log::DebugLog::DebugLog(): error: undefined reference to 'vtable for modsecurity::debug_log::DebugLog'
/usr/bin/ld.gold: the vtable symbol may be undefined because the class is missing its key function
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function modsecurity::RulesProperties::RulesProperties(modsecurity::debug_log::DebugLog*): error: undefined reference to 'modsecurity::audit_log::AuditLog::AuditLog()'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function modsecurity::RulesProperties::RulesProperties(modsecurity::debug_log::DebugLog*): error: undefined reference to 'modsecurity::RulesExceptions::RulesExceptions()'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function modsecurity::RulesProperties::RulesProperties(modsecurity::debug_log::DebugLog*): error: undefined reference to 'modsecurity::RulesExceptions::~RulesExceptions()'
bazel-out/k8-fastbuild/bin/http-filter-modsecurity/_objs/http_filter_lib/http_filter.pic.o:http_filter.cc:function modsecurity::RuleMessage::log[abi:cxx11](modsecurity::RuleMessage const*, int): error: undefined reference to 'modsecurity::RuleMessage::log[abi:cxx11](modsecurity::RuleMessage const*, int, int)'
collect2: error: ld returned 1 exit status
Target //http-filter-modsecurity:envoy failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 11.234s, Critical Path: 4.27s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

**Comment by rbkumar88** [comment]
Facing same issue..@ekarlso were you able to resolve this?

**Comment by moderation** [comment]
Envoy just released [1.10.0](https://www.envoyproxy.io/docs/envoy/v1.10.0/intro/version_history#apr-5-2019). 1.7.1 is very old and the there have been many deprecations and changes since then. It looks like this filter was thrown over the wall at a point in time and not maintained since.

**Comment by yoshpe** [comment]
I've updated Envoy to 1.11.0
